<a href="https://colab.research.google.com/github/davidms22/UFV_Visualizacion/blob/main/Ejercicios%20Clase/Clase_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**EJERCICIO CLASE 18/10/2024**

Para este ejercicio, vamos a trabajar con la base de datos *'datos_ejercicios_ventas.csv'*.

En primer lugar, abrimos la base de datos con excel para poder observarla y entenderla. En este caso debemos aprovechar que es de un tamaño razonable para entenderla al completo.

- **COUNTRY**: País donde se registran las ventas.
- **SUBBRAND**: Submarca del producto involucrado en las ventas.
- **YEAR**: Año del registro de ventas o predicciones.
- **MONTH**: Mes correspondiente al registro.
- **SCENARIO**: Indicador de observacion real o predicha con AI.
- **FORECAST**: Código que identifica la predicción realizada (ej. "AI_P02F").
- **FORECAST_YEAR**: Año asociado a la predicción.
- **AMOUNT**: Valor numérico que representa el monto de ventas en unidades

In [31]:
# Cargamos base de datos y mostramos primera fila.
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv('datos_ejercicio_ventas.csv')
print(df.head())

         COUNTRY        SUBBRAND  YEAR  MONTH     SCENARIO FORECAST  \
0       Portugal     Lipton (L3)  2023     12  AI_forecast  AI_P02F   
1  Great Britain     Lipton (L3)  2023     12  AI_forecast  AI_P10F   
2          Spain  Pepsi Max (L3)  2023     12  AI_forecast  AI_P09F   
3  Great Britain        7up (L3)  2024     12  AI_forecast  AI_P10F   
4        Hungary     Lipton (L3)  2023      9  AI_forecast  AI_P03F   

   FORECAST_YEAR         AMOUNT  
0         2023.0  754356.237194  
1         2023.0  560030.558029  
2         2023.0   88501.980847  
3         2023.0  363224.511516  
4         2023.0  396176.120491  


Al ver la primera fila, podemos hacer el ejercicio de traducirla al lenguaje y comprobar si hemos entendido la base de datos.

**En este caso**: en enero de 2023, se realizó una prediccion con IA para obtener la cantidad de Lipton vendido en Portugal durante el mes de diciembre de 2023.

Una vez entendida la base de datos, estamos preparados para poder trabajar con ella.

**PREGUNTAS QUE RESPONDER.**

1. Como se distribuyen las ventas realizadas en:
    - Cada país.
    - Cada mes y año.
    - Cada marca.

2. Cual es la tendencia y estacionalidad de:
    - Todas las ventas del país con menos ventas.
    - La marca con más ventas.

3. Cuales son las predicciones hechas en España y como de
buenas son.

In [39]:
# Vemos el tamaño de la base de datos.
print(f"El tamaño de la base de datos es: {df.shape}")

# Vemos la cantidad de observaciones diferenciadas en 'actual' o 'forecast'
observations_count = df['SCENARIO'].value_counts()
print(observations_count)

El tamaño de la base de datos es: (18666, 8)
SCENARIO
AI_forecast    17766
actual           900
Name: count, dtype: int64


La base de datos tiene 18666 observaciones de las cuales 900 son valores actuales y las demás son predicciones.

In [38]:
# Mapeamos FORECAST para que en vez del codigo, me salga el mes en el que se hizo la predicción.
month_mapping = {
    "AI_P02F": "enero",
    "AI_P03F": "febrero",
    "AI_P04F": "marzo",
    "AI_P05F": "abril",
    "AI_P06F": "mayo",
    "AI_P07F": "junio",
    "AI_P08F": "julio",
    "AI_P09F": "agosto",
    "AI_P10F": "septiembre",
    "AI_P11F": "octubre",
    "AI_P12F": "noviembre",
    "AI_P13F": "diciembre"
}

df['FORECAST'] = df['FORECAST'].replace(month_mapping)
print(df.head())

         COUNTRY        SUBBRAND  YEAR  MONTH     SCENARIO    FORECAST  \
0       Portugal     Lipton (L3)  2023     12  AI_forecast       enero   
1  Great Britain     Lipton (L3)  2023     12  AI_forecast  septiembre   
2          Spain  Pepsi Max (L3)  2023     12  AI_forecast      agosto   
3  Great Britain        7up (L3)  2024     12  AI_forecast  septiembre   
4        Hungary     Lipton (L3)  2023      9  AI_forecast     febrero   

   FORECAST_YEAR         AMOUNT  
0         2023.0  754356.237194  
1         2023.0  560030.558029  
2         2023.0   88501.980847  
3         2023.0  363224.511516  
4         2023.0  396176.120491  


In [37]:
# Verificamos si hay NaNs.
print(df.isnull().sum())

COUNTRY            0
SUBBRAND           0
YEAR               0
MONTH              0
SCENARIO           0
FORECAST         900
FORECAST_YEAR    900
AMOUNT             0
dtype: int64


Como podemos observar, hay 900 NaNs en 'FORECAST' y 'FORECAST_YEAR', que representan las 900 observaciones las cuales no son predicciones si no valores reales.

In [35]:
# Número de paises
num_paises = df['COUNTRY'].nunique()
print(f"Número total de paises: {num_paises}")

paises = df['COUNTRY'].unique()
print(f"Los paises son: {paises}")

# Número sde productos.
num_prod = df['SUBBRAND'].nunique()
print(f"Número total de productos: {num_prod}")

productos = df['SUBBRAND'].unique()
print(f"Los productos son: {productos}")

Número total de paises: 9
Los paises son: ['Portugal' 'Great Britain' 'Spain' 'Hungary' 'Norway' 'Denmark'
 'Netherlands' 'Italy' 'Czech']
Número total de productos: 6
Los productos son: ['Lipton (L3)' 'Pepsi Max (L3)' '7up (L3)' 'Pepsi Regular (L3)'
 'Mountain Dew (L3)' '7up Free (L3)']
